modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -13,45 +13,152 @@
 
 """Module houses `ParquetDispatcher` class, that is used for reading `.parquet` files."""
 
+from abc import ABC, abstractmethod, abstractproperty
 import os
 
+import json
 import fsspec
 from fsspec.core import split_protocol, url_to_fs
 from fsspec.registry import get_filesystem_class
 from fsspec.spec import AbstractBufferedFile
 import numpy as np
-from pandas.io.common import is_fsspec_url
 from packaging import version
 
 from modin.core.storage_formats.pandas.utils import compute_chunksize
 from modin.config import NPartitions
 
 
 from modin.core.io.column_stores.column_store_dispatcher import ColumnStoreDispatcher
-from modin.utils import import_optional_dependency
+from modin.utils import import_optional_dependency, _inherit_docstrings
+
+
+class Dataset(ABC):  # noqa : PR01
+    """
+    Abstract class that encapsulates Parquet engine-specific details for
+    a given Parquet dataset.
+
+    This class exposes a set of functions that are commonly used in the
+    `read_parquet` implementation.
+
+    Attributes
+    ----------
+    path : str, path object or file-like object
+        The filepath of the parquet file in local filesystem or hdfs.
+    storage_options : dict
+        Parameters for specific storage engine.
+    _fs_path : str, path object or file-like object
+        The filepath or handle of the parquet dataset specific to the
+        filesystem implementation. E.g. for `s3://test/example`, _fs
+        would be set to S3FileSystem and _fs_path would be `test/example`.
+    _fs : Filesystem
+        Filesystem object specific to the given parquet file/dataset.
+    dataset : ParquetDataset or ParquetFile
+        Underlying dataset implementation for PyArrow and fastparquet
+        respectively.
+    _row_groups_per_file : list
+        List that contains the number of row groups for each file in the
+        given parquet dataset.
+    _files : list
+        List that contains the full paths of the parquet files in the dataset.
+    """
+
+    def __init__(self, path, storage_options):
```

Comment:
An abstract class shouldn't be instantiated. We do use such a logic nowhere. I suggest using an ordinary python class instead of abc and raise NotImplemented error for the methods of this class that has to be overridden.

Reply:
Yes you're right. Initially it made sense to make use of the abstract class but it's at a point where it doesn't make sense to do that anymore.